lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2978,17 +2978,24 @@ static bool checkDifferentiationParameters(
   SmallVector<Type, 4> wrtParamTypes;
   indices->getSubsetParameterTypes(functionType, wrtParamTypes);
   for (unsigned i : range(wrtParamTypes.size())) {
+    SourceLoc loc = parsedWrtParams.empty()
+        ? attrLoc
+        : parsedWrtParams[i].getLoc();
     auto wrtParamType = wrtParamTypes[i];
+    if (wrtParamType->is<InOutType>()) {
+      TC.diagnose(
+          loc,
+          diag::diff_params_clause_inout_argument,
```

Comment:
I think it would make sense to add a test to make sure that this diagnose is correctly triggered, wdyt?

Reply:
sorry ... I see you already attempted to do so https://github.com/apple/swift/pull/25715#issuecomment-505206714

Reply:
Yes although it’s not triggered right now and the test does not pass. I haven’t yet figured out why.